pkg/drainevent/scheduled-event.go

```diff
@@ -20,54 +20,62 @@ const (
 )
 
 // MonitorForScheduledEvents continuously monitors metadata for scheduled events and sends drain events to the passed in channel
-func MonitorForScheduledEvents(drainChan chan<- DrainEvent, cancelChan chan<- DrainEvent, nthConfig config.Config) {
-	log.Println("Started monitoring for scheduled events")
-	for range time.Tick(time.Second * 2) {
-		drainEvents := CheckForScheduledEvents(nthConfig.MetadataURL)
-		for _, drainEvent := range drainEvents {
-			if !isStateCancelledOrCompleted(drainEvent.State) {
-				log.Println("Sending drain events to the drain channel")
-				drainChan <- drainEvent
-				// cool down for the system to respond to the drain
-				time.Sleep(120 * time.Second)
-			} else if isStateCancelledOrCompleted(drainEvent.State) {
-				log.Println("Sending cancel events to the cancel channel")
-				cancelChan <- drainEvent
-			}
+func MonitorForScheduledEvents(drainChan chan<- DrainEvent, cancelChan chan<- DrainEvent, nthConfig config.Config) error {
+	drainEvents, err := checkForScheduledEvents(nthConfig.MetadataURL)
+	if err != nil {
+		return err
+	}
+	for _, drainEvent := range drainEvents {
+		if isStateCancelledOrCompleted(drainEvent.State) {
+			log.Println("Sending cancel events to the cancel channel")
+			cancelChan <- drainEvent
+		} else {
+			log.Println("Sending drain events to the drain channel")
+			drainChan <- drainEvent
+			// cool down for the system to respond to the drain
+			time.Sleep(120 * time.Second)
 		}
 	}
+	return nil
 }
 
-// CheckForScheduledEvents Checks EC2 instance metadata for a scheduled event requiring a node drain
-func CheckForScheduledEvents(metadataURL string) []DrainEvent {
-	events := make([]DrainEvent, 0)
+// checkForScheduledEvents Checks EC2 instance metadata for a scheduled event requiring a node drain
+func checkForScheduledEvents(metadataURL string) ([]DrainEvent, error) {
 	resp, err := ec2metadata.RequestMetadata(metadataURL, ec2metadata.ScheduledEventPath)
 	if err != nil {
-		log.Fatalf("Unable to parse metadata response: %s", err.Error())
+		return nil, fmt.Errorf("Unable to parse metadata response: %w", err)
```

Comment:
note that `%w` is go 1.13+. Is NTH only built on 1.13 and up? If so, cool. if not, well, prolly want to use `%v` or `%s` here.

Reply:
Alternately, you could always return an `errors.Wrapf()` here instead?

https://godoc.org/github.com/pkg/errors#Wrapf

Reply:
NTH is only built on 1.13 (the binary distributions and docker images). I suppose the only concern is if someone is building from source and making their own distribution images. `%w` is used elsewhere in the codebase, so I'll go ahead and merge this one. We can chat if this is a big issue or not.